packages/freezed/lib/src/freezed_generator.dart

```diff
@@ -174,6 +175,8 @@ class FreezedGenerator extends ParserGenerator<_GlobalData, Data, Freezed> {
 
     return _computeElementDataCache.putIfAbsent(classElement, () {
       if (!typeChecker.hasAnnotationOf(classElement)) return null;
+      if (_parsedElementCheckList.contains(classElement)) return null;
```

Comment:
That "return null" sounds fishy.

Maybe we should replace the putIfAbsent with a cache[key] = whatever; instead. And do nothing in this situation (whereas the other line actually assign null)

Reply:
so something like this ?
```dart    
if (!typeChecker.hasAnnotationOf(classElement) || _parsedElementCheckSet.contains(classElement)) {
  return null;
} else {
  _parsedElementCheckSet.add(classElement);
  final data = parseElement(
    _GlobalData(hasDiagnostics: false, hasJson: false),
    classElement,
  );
  _computeElementDataCache[classElement] = data;
  return data;
}
```